src/core/net/tcp6.cpp

```diff
@@ -68,67 +77,131 @@ Error Tcp::Endpoint::Initialize(Instance &aInstance, otTcpEndpointInitializeArgs
     mReceiveAvailableCallback = aArgs.mReceiveAvailableCallback;
     mDisconnectedCallback     = aArgs.mDisconnectedCallback;
 
-    mInstance = &aInstance;
-
     memset(mTimers, 0x00, sizeof(mTimers));
+    memset(&mSockAddr, 0x00, sizeof(mSockAddr));
+    memset(tp, 0x00, sizeof(struct tcpcb));
+
+    /*
+     * Initialize buffers --- formerly in initialize_tcb.
+     */
+    {
+        uint8_t *recvbuf    = static_cast<uint8_t *>(aArgs.mReceiveBuffer);
+        size_t   recvbuflen = aArgs.mReceiveBufferSize - ((aArgs.mReceiveBufferSize + 8) / 9);
+        uint8_t *reassbmp   = recvbuf + recvbuflen;
+
+        lbuf_init(&tp->sendbuf);
+        cbuf_init(&tp->recvbuf, recvbuf, recvbuflen);
+        tp->reassbmp = reassbmp;
+        bmp_init(tp->reassbmp, BITS_TO_BYTES(recvbuflen));
+    }
+
+    tp->accepted_from = nullptr;
+    initialize_tcb(tp);
+
+    /* Note that we do not need to zero-initialize mReceiveLinks. */
+
+    tp->instance = &aInstance;
 
 exit:
     return error;
 }
 
 Instance &Tcp::Endpoint::GetInstance(void)
 {
-    return AsCoreType(mInstance);
+    struct tcpcb *tp = &GetTcb();
+
+    return AsCoreType(tp->instance);
```

Comment:
```suggestion
    struct tcpcb &tp = &GetTcb();

    return AsCoreType(tp.instance);
```

nit: prefer alias to pointer?

Reply:
I made this change throughout the file (changing all `struct tcpcb *tp = &GetTcb();` into `struct tcpcb &tp = GetTcb();` and all uses of `tp` as needed).